src/mds/MDSRank.cc

```diff
@@ -1800,9 +1793,15 @@ class C_MDS_Send_Command_Reply : public MDSInternalContext
  */
 void MDSRankDispatcher::evict_sessions(const SessionFilter &filter, MCommand *m)
 {
-  std::list<Session*> victims;
   C_MDS_Send_Command_Reply *reply = new C_MDS_Send_Command_Reply(this, m);
 
+  if (is_any_replay()) {
+    reply->send(-EINVAL, "MDS is replaying log");
```

Comment:
Let's return -EAGAIN instead, so that a caller can tell the difference between this case and the case where their filter expression was invalid